pages/onboarding/index.tsx

```diff
@@ -1,15 +1,131 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React, { useState, useEffect } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface Props {
+  user: UserProfile | null;
+}
+
+const OnboardingPage: NextPage<Props> = ({ user }: Props) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: user?.bio,
+    twitter_username: user?.twitter_username,
+    linkedin_url: user?.linkedin_url,
+    github_username: user?.github_username,
+    website: user?.website,
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFormFilled, setIsFormFilled] = useState(false);
+
+  const router = useRouter();
+
+  useEffect(() => {
+    if (!user) {
+      router.push('/');
+    }
+  }, [user]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    const isFormFilled = Object.values(formData).some(value => !!value);
+    setIsFormFilled(isFormFilled);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      setIsLoading(true);
+      await axios.patch(`/api/users/${user?.id}`, formData);
+      setIsLoading(false);
+      router.push('/directory');
+    } catch (error) {
+      setIsLoading(false);
+      alert(error);
+    }
+  };
+
+  const isSubmitDisabled = !isFormFilled && !isLoading;
```

Comment:
i'm not sure if this logic is what we want - i'm reading this as: "submit is disabled if the form is not filled, and it is not loading" - meaning:
* if `isLoading` = true, then `isSubmitDisabled` = false

we can break this down into thinking: when do we want the button to be disabled?
* when the form is not filled (nothing to submit)
* when they're in the middle of a submission (avoid double submits)

these two criteria can be respectively expressed as:
* `!isFormFilled` (which you have), OR
* `isLoading`

```suggestion
  const disableSubmit = !isFormFilled || isLoading;
```